### PR TITLE
re-enable unit tests

### PR DESCRIPTION
**Pull request by myk002**
and fix up the issues we find

### CODE REVIEW

File: test/gui.lua [hunk 1]
```diff
@@ -1,3 +1,5 @@
+config.targets = 'core'
```
Comment (lethosor) [inline test/gui.lua:1]: Are we planning to support multiple targets per test? Would suggest `target` if not.

Comment (myk002) [reply test/gui.lua:1]: I had the quickfort suite of integration tests in mind when I made this plural. I could also make a singular variant of the property. The overwhelming majority of tests test one target (being "unit" tests and all). Maybe calling it "target" but still letting it take a list for integration cases is the way to go.

File: ci/test.lua [hunk 2]
```diff
@@ -379,7 +379,7 @@ local function load_tests(file, tests)
         dfhack.printerr('Skipping tests for unspecified target in ' .. file)
         return true  -- TODO: change to false once existing tests have targets specified
     end
-    local targets = type(env.config.targets) == table and env.config.targets or {env.config.targets}
+    local targets = type(env.config.target) == table and env.config.target or {env.config.target}
```
Comment (lethosor) [inline ci/test.lua:382]: `type()` returns strings:
```suggestion
    local targets = type(env.config.target) == 'table' and env.config.target or {env.config.target}
```

Comment (myk002) [reply ci/test.lua:382]: ha whoops. thanks. fixed.

**Pull request merged**